frontend/src/lib/codemirror/useCompareExtension.ts

```diff
@@ -88,38 +45,55 @@ const diffGutter = gutter({
         }
     },
     lineMarkerChange(update) {
-        return update.docChanged || !compareNullableText(update.state.facet(targetText), update.startState.facet(targetText))
+        return update.docChanged || (update.state.facet(diffLineMap) != update.startState.facet(diffLineMap))
     },
     initialSpacer: () => marker,
 })
 
-export function useDelayedCompareExtension(viewRef: RefObject<EditorView>, compareTo: string, delayMs = 1000): Extension {
-    const editTime = useRef(0)
-    const timeSinceEdit = Date.now() - editTime.current
-    const timeout = useRef<any>()
-    const [, forceUpdate] = useState({})
-
-    return [
-        useCompareExtension(viewRef, timeSinceEdit > delayMs ? compareTo : undefined),
-        EditorState.transactionExtender.of(tr => {
-            if (tr.docChanged) {
-                editTime.current = Date.now()
-
-                if (timeout.current)
-                    clearTimeout(timeout.current)
-                timeout.current = setTimeout(() => {
-                    forceUpdate({})
-                }, delayMs)
-            }
-
-            return null
-        }),
-    ]
-}
+const createDiffWorker = createWorkerFactory(() => import("./useCompareExtension.worker"))
 
 // Extension that highlights lines in the doc that differ from `compareTo`.
 export default function useCompareExtension(viewRef: RefObject<EditorView>, compareTo: string): Extension {
     const [compartment] = useState(new Compartment())
+    const [diffLineMapCompartment] = useState(new Compartment())
+
+    const plugin = ViewPlugin.fromClass(class {
```

Comment:
Might be better to hoist the class to the top level and give it a name. This anonymous class confused me for a sec.

Reply:
will do